src/main/java/org/ice4j/ice/Component.java

```diff
@@ -902,6 +922,12 @@ public RemoteCandidate findRemoteCandidate(TransportAddress remoteAddress)
      */
     protected void setSelectedPair(CandidatePair pair)
     {
+        if (keepAliveStrategy == KeepAliveStrategy.SELECTED_ONLY)
```

Comment:
I check the state of the Agent instead, and throw an exception if it's been started. Would have had to add parameters in 3 places otherwise.